homeassistant/components/here_travel_time/__init__.py

```diff
@@ -108,171 +93,25 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
     return unload_ok
 
 
-class HereTravelTimeDataUpdateCoordinator(DataUpdateCoordinator):
-    """HERETravelTime DataUpdateCoordinator."""
-
-    def __init__(
-        self,
-        hass: HomeAssistant,
-        api: RoutingApi,
-        config: HERETravelTimeConfig,
-    ) -> None:
-        """Initialize."""
-        super().__init__(
-            hass,
-            _LOGGER,
-            name=DOMAIN,
-            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
-        )
-        self._api = api
-        self.config = config
-
-    async def _async_update_data(self) -> HERERoutingData | None:
-        """Get the latest data from the HERE Routing API."""
-        try:
-            async with async_timeout.timeout(10):
-                return await self.hass.async_add_executor_job(self._update)
-        except NoRouteFoundError as error:
-            raise UpdateFailed(NO_ROUTE_ERROR_MESSAGE) from error
-
-    def _update(self) -> HERERoutingData | None:
-        """Get the latest data from the HERE Routing API."""
-        try:
-            origin, destination, arrival, departure = self._prepare_parameters()
+async def _async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
 
-            _LOGGER.debug(
-                "Requesting route for origin: %s, destination: %s, route_mode: %s, mode: %s, traffic_mode: %s, arrival: %s, departure: %s",
-                origin,
-                destination,
-                RouteMode[self.config.route_mode],
-                RouteMode[self.config.travel_mode],
-                RouteMode[TRAFFIC_MODE_ENABLED],
-                arrival,
-                departure,
-            )
+    if config_entry.version == 1:
```

Comment:
My impression is that there are some downsides to needing to perform migration, and if you can just update w/o setting versions based on fields then its better to do that. (I can't remember exactly the details of the side effects that migration causes).  Can you do this without updating versions?

Reply:
I don't understand how migrations are supposed to work and I also found different ways of doing them in other integrations. The way as it is described [in the dev docs](https://developers.home-assistant.io/docs/config_entries_config_flow_handler?_highlight=migrate#config-entry-migration) did not work for me.

I don't see a need for changing the version, I was cargo-culting here. I only need
```python
new_data = {**config_entry.data}
        if new_data[CONF_MODE] == "publicTransportTimeTable":
            new_data[CONF_MODE] = TRAVEL_MODE_PUBLIC

        new_options = {**config_entry.options}
        if "traffic_mode" in new_options:
            new_options.pop("traffic_mode")
```

Reply:
I remember know where I had feedback on my own use of this:
https://github.com/home-assistant/core/pull/44202#discussion_r546294044

```
async_migrate_entry should only be used when we need to change the entry data, it can be done automatically and it's not possible to avoid backwards incompatible changes of the entry data.

Most entry changes eg adding entry unique id can be done without migration allowing users to roll back if needed.
```

So it sounds like my specific case was because i was just adding something that could easily be rolled back with no other changes -- so ignore what I said and consider this in the case of ability to rollback.

My read is that this is removing old unused fields that don't really make sense in the new flow, so another angle is to just ignore them to support clean rollback.

Reply:
We could say that `publicTransportTimeTable` will be handled as `publicTransport` internally.
We could also ignore the `traffic_mode` setting as is already the case.

The downside I see is that this might confuse users as there are things in the front-end which have no function or a different function behind the scenes.

In regards to avoiding the breaking change I also want to highlight that the user does not have to do anything and everything _should_ behave as before. But as we are changing the API this integration uses we cannot guarantee it. Traffic might be calculated a bit differently etc.

Reply:
Changing the config flow makes sense to me, so I think this is just about the storage and supporting both old and new formats?

Reply:
That is true. If I remove the migration I would have to implement backwards compatible logic to init and config_flow.

Do you think removing the migration outweighs more complicated logic in the rest of the integration?

Reply:
What logic has to happen elsewhere? I read this as checking `CONF_MODE` once on start, but I may have misunderstood.

Reply:
I pushed a commit with the alternative implementation